### ISSUE TITLE
`manual_str_repeat` false positive when repeating a char

### ISSUE THREAD

**Issue by mcobzarenco** [rust-lang/rust-clippy#7750]
Lint name: `manual_str_repeat`

I tried this code:

```rust
iter::repeat('─').take(10).collect::<String>()
```

Which trips up `manual_str_repeat`, although I don't think this is correct -- or at least the suggested fix, `"─".repeat(10)` doesn't work, as I'm repeating a `char`, not an `str`. Note 
 - there's no `repeat` function defined on a char
 - repeating a `str` is not semantically the same; e.g. in my example it's not possible to collect it into a `String`

I expected to see no clippy warning.

```
warning: manual implementation of `str::repeat` using iterators
    |
150 |                            iter::repeat('─')
    |  __________________________^
151 | |                             .take(10)
152 | |                             .collect::<String>(),
    | |________________________________________________^ help: try this: `"─".repeat(10)`
    |
    = note: `#[warn(clippy::manual_str_repeat)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_str_repeat
```

### Meta

**Rust version (`rustc -Vv`):**
```
rustc 1.55.0 (c8dfcfe04 2021-09-06)
binary: rustc
commit-hash: c8dfcfe046a7680554bf4eb612bad840e7631c4b
commit-date: 2021-09-06
host: x86_64-unknown-linux-gnu
release: 1.55.0
LLVM version: 12.0.1
```

@rustbot label +I-suggestion-causes-error

**Comment by mikerite** [comment]
Replace the whole expression with `"─".repeat(10)`. You don't need to collect to `String` because `str::repeat` returns a `String`.

[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=0e46543087441e297377c6a4e8bcf1ee)

**Comment by mikerite** [comment]
@dswij It's not valid. Please give an actual code example where the (correctly applied) suggestion either gives a compiler error or a different result. The values returned are exactly the same.
